### DISCUSSION TITLE
Automatizes any edf meter for 20€ in 5 minutes - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#47] [Comments]
# Automatizes any edf meter for 20€ in 5 minutes - Haade.fr

Reads the teleinfo in 5 minutes in home assistant on any edf meter (old and linky) with a teleinfo v3.0 micro key by charles Hallard at 20€, via the mqtt and teleinfo2mqtt modules

https://haade.fr/en/blog/domotics-linky-%C2%B5teleinfo-homeassistant-20-5minutes

**Comment by Jpsy** [comment]
Hi Nico, 
thanks for your great guide. This really helped my a lot to set up my Pi to follow my Linky consumption in Home Assistant. The Micro Teleinfo V3.0 really is a great module which worked immediately.

May I still suggest two little changes to your guide?
- In the configuration of teleinfo2mqtt you can actually use the much shorter device path (like "/dev/ttyUSB0" or /dev/ttyACM0") of the serial port, instead for the full ID (like "/dev/serial/by-id/usb-CH2i_uTinfo-V3.0_TINFO-2046-if00"). Both can be found in the hardware section that you describe in the guide.
- The gauge should actually not have the title "Volt". It should rather be "Apparent power" ("Puissance apparente"  in French).

Thanks again!
Jörg

**Reply by haade-administrator** [reply]
Bonjour Jpsy /dev/ttyUSB0 ou /dev/ttyACM0 est à éviter en production si tu as plusieurs clés de branchés ton pc peut modifier la sortie après un démarrage par exemple comme /dev/ttyUSB1, alors que by-id te garantie la bonne clé c'est très conseillé.

En ce qui concerne le relevé de puissance il se fait en VA ( volt ampère ) chez EDF

Hello Jpsy /dev/ttyUSB0 or ​​/dev/ttyACM0 should be avoided in production if you have several keys connected your PC can modify the output after a start for example like /dev/ttyUSB1, while by-id guarantees you the right key it is very recommended.

Regarding the power reading it is done in VA (volt ampere) at EDF
